### PR TITLE
Added basic mocha test

### PR DESCRIPTION
**Pull request by cgole**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,8 +6,10 @@
   "directories": {
     "example": "examples"
   },
-  "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+   "scripts": {
+    "install_client": "npm i client/",
+    "test": "mocha -R spec test",
```
Comment (rmg) [inline package.json]: This should be `test/*test.js` to make sure it doesn't try to run the test/start-server.js as a test.

**Comment by rmg** [comment]
Should have a `.jshintignore` in the root with `node_modules/` in it.

**Comment by cgole** [comment]
@superkhau  please review

**Comment by superkhau** [comment]
Why the commented tests?

File: test/rest_api_test.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    done();
+  });
+  
+  after(function(done) {
+    app.removeAllListeners('started');
+    app.removeAllListeners('loaded');
+    done();
+  });
+  
+  // it('should not allow access without access token', function(done){
```
Comment (superkhau) [inline test/rest_api_test.js]: If you want to keep the tests around. I recommend using `it.skip('should...` instead. In addition, you should leave a comment stating why this test was skipped. Same goes for the other commented tests below.

Comment (cgole) [inline test/rest_api_test.js]: oops. they were not supposed to be there. deleted it

**Comment by superkhau** [comment]
LGTM now. ;)

**Pull request merged**